### PR TITLE
web/ui: initial commit of Flow UI

### PR DESCRIPTION
**Pull request by rfratto**
This is the initial commit of the Flow UI. It has two views:

* A component table view 
* A component detail view, to debug individual components 

This initial commit does not yet wire the UI into the Grafana Agent binary. This will happen in a future PR.

To run locally, do the following:

* Run Grafana Agent Flow 
* In another tab, run `yarn` in the `web/ui` folder.
* Then run `yarn start` in the `web/ui` folder.
* Navigate to http://localhost:3000 to access the UI. 

You must have the following installed to run the UI locally:

* Yarn >= v1.22
* Node.js >= v16

### Component table view

image

### Component detail view 

image

### CODE REVIEW

File: web/ui/.env.production [hunk 1]
```diff
@@ -0,0 +1,2 @@
+REACT_APP_BASE_URL={{! .PublicURL !}}
+PUBLIC_URL={{! .PublicURL !}}/public
```
Comment (rfratto) [inline web/ui/.env.production:2]: I need to add a comment explaining this syntax

**Comment by rfratto** [comment]
I've temporarily removed the graph view from the PR to make this easier to review; I'll add the graph view back in another PR.

**Review by mattdurham — APPROVED** [review]
LGTM

I am by no means a typescript or front-end developer so mainly glanced at the code to make sure nothing stood out and that I could follow what it was doing without getting into heavy detail. I then also ran the UI against main and ensure that the front end was rendered and displayed correctly.

**Review by jcreixell — APPROVED** [review]
From my limited react experience LGTM, looks quite neat

**Review by rlankfo — APPROVED** [review]
LGTM +1

**Comment by rfratto** [comment]
I'm going to merge for now. The code probably needs more love long-term, but I feel it's in a maintainable state for an initial commit.

**Pull request merged**